great_expectations/datasource/fluent/data_asset/data_connector/s3_data_connector.py

```diff
@@ -22,6 +24,12 @@
 logger = logging.getLogger(__name__)
 
 
+class _S3Options(pydantic.BaseModel):
```

Comment:
@Kilo59 One of the gaps in the current implementation that has not yet been addressed is that there could be other access options supported by each of these file storage services (different ones, respectively), and we are not including them.  I think that it is accepted for now to support the ones we do; we can expand based on user inquiries (or contributors can add them).